src/it/scala/ApiServiceISpec.scala

```diff
@@ -0,0 +1,146 @@
+    jmsMessage.asTextF[IO].attempt.map {
+      case Left(e) => fail(s"Unable to read message file due to ${e.getMessage}")
+      case Right(text) =>
+        val file = Paths.get(replyMessage)
+        Using(Files.newBufferedWriter(file)) { writer =>
+          writer.write(text)
+        } match {
+          case Success(_) => ()
+          case Failure(e) => fail(s"Unable to write message file due to ${e.getMessage}")
+        }
+    }
+
+  def getFileContent: IO[String] =
+    IO.pure(getMessageFromFileStringPath(replyMessage))
+
+  def getMessageFromFileStringPath(messageFilePath: String): String =
+    Using(Source.fromFile(messageFilePath)) { source =>
+      source.getLines().mkString
+    } match {
+      case Success(value) => value
+      case Failure(_)     => ""
+    }
+
+  "The Message API" - {
+
+    "returns an echo message when a text message is sent with a SID of ECHO001" in {
+      val sqsTestConnector = SqsConnector("http://localhost:9324")
+      val sqsTestConnection: Connection = sqsTestConnector.getConnection
+      val session: Session = sqsTestConnection.createSession(false, Session.AUTO_ACKNOWLEDGE)
+      val searchQueue: String = "request-general"
+      val producer: MessageProducer = session.createProducer(session.createQueue(searchQueue))
+
+      val textMessage = session.createTextMessage("Hello World!")
+      textMessage.setStringProperty("sid", "ECHO001")
+      producer.send(textMessage)
+      eventually {
+        getFileContent.asserting(_ mustBe "The Echo Service says: Hello World!")
+      }
+    }
+
+    "returns an error message when the SID is not recognised" in {
+      pending // This test is marked pending until completion of https://national-archives.atlassian.net/browse/PACT-836
+      val sqsTestConnector = SqsConnector("http://localhost:9324")
+      val sqsTestConnection: Connection = sqsTestConnector.getConnection
+      val session: Session = sqsTestConnection.createSession(false, Session.AUTO_ACKNOWLEDGE)
+      val searchQueue: String = "request-general"
+      val producer: MessageProducer = session.createProducer(session.createQueue(searchQueue))
+
+      val textMessage = session.createTextMessage("Hello World!")
+      textMessage.setStringProperty("sid", "UNKNOWN")
+      producer.send(textMessage)
+      eventually {
+        getFileContent.asserting(_ mustBe "The Echo Service says: Hello World!")
+      }
+    }
+
+    "returns an error message when the message body is empty" in {
+      pending // This test is marked pending until completion of https://national-archives.atlassian.net/browse/PACT-836
+      val sqsTestConnector = SqsConnector("http://localhost:9324")
+      val sqsTestConnection: Connection = sqsTestConnector.getConnection
+      val session: Session = sqsTestConnection.createSession(false, Session.AUTO_ACKNOWLEDGE)
+      val searchQueue: String = "request-general"
+      val producer: MessageProducer = session.createProducer(session.createQueue(searchQueue))
```

Comment:
If we could send a unique message ID here, then we could have corresponding reply, exclusive for this scenario.

Reply:
This is outside of our control. In our request-reply client [we delegate this responsibility to jms4s](https://github.com/nationalarchives/ctd-omega-editorial-frontend/blob/148377659968f23d6c03820556a8a4f7b66a6e89/app/uk/gov/nationalarchives/omega/editorial/connectors/JmsRequestReplyClient.scala#L73) which in turn delegates it to SQS. We get the `messageId` back and store in in a map but we don't create it.

Reply:
Understood that we don't choose the message ID.